pkgs/tools/filesystems/f3/default.nix

```diff
@@ -1,38 +1,54 @@
-{ stdenv, fetchFromGitHub
-, parted, udev
+{ stdenv, lib, fetchFromGitHub
+, parted, systemd ? null
 }:
 
 stdenv.mkDerivation rec {
   pname = "f3";
   version = "7.2";
 
-  enableParallelBuilding = true;
-
   src = fetchFromGitHub {
     owner = "AltraMayor";
     repo = pname;
     rev = "v${version}";
     sha256 = "1iwdg0r4wkgc8rynmw1qcqz62l0ldgc8lrazq33msxnk5a818jgy";
   };
 
-  buildInputs = [ parted udev ];
+  postPatch = ''
+     sed -i 's/-oroot -groot//' Makefile
+
+     for f in f3write.h2w log-f3wr; do
+      substituteInPlace $f \
+        --replace '$(dirname $0)' $out/bin
+     done
+  '';
+
+  buildInputs = [
+    parted
+  ]
+  ++ lib.optional stdenv.isLinux systemd;
+
+  enableParallelBuilding = true;
 
-  patchPhase = "sed -i 's/-oroot -groot//' Makefile";
+  buildFlags   = [
+    "all" # f3read, f3write
+  ]
+  ++ lib.optional stdenv.isLinux "extra"; # f3brew, f3fix, f3probe
 
-  buildFlags   = [ "all"                    # f3read, f3write
-                   "extra"                  # f3brew, f3fix, f3probe
-                 ];
+  installFlags = [
+    "PREFIX=$(out)"
+    "install"
+  ]
+  ++ lib.optional stdenv.isLinux "install-extra";
```

Comment:
These should go to `installTargets`, otherwise we have `make install PREFIX=$(out) install install-extra`.